dao/secret_dao.go

```diff
@@ -0,0 +1,121 @@
+package dao
+
+import (
+	"context"
+	"fmt"
+
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+	"gorm.io/gorm"
+)
+
+const SecretResourceType = "Tenant"
+
+var GetSecretDao func(daoParams *RequestParams) SecretDao
+
+type secretDaoDbImpl struct {
+	TenantID *int64
+	UserID   *int64
```

Comment:
this is kind of a general thing - I was wondering if maybe in the future we should just embed the reqestParams struct in the impl - that way we don't have to copy the fields in the `get**` functions.

Reply:
that would work - great idea - let's do in other PR.